### PR TITLE
Use base class method to get valid fields for type

### LINKED ISSUE #260: NewRowSynthesis should ignore PrimaryKey column
### Problem Description

It appears that the `NewRowSynthesis` measure includes the `primary_key` column in its analysis. I can feed it two tables that are identical except for the `primary_key` column, and it will measure every synthetic data row as new even though practically speaking none of them are new.

### Expected behavior

Ignore the `primary_key` column when doing `NewRowSynthesis`.

### PR DESCRIPTION
**Pull request by katxiao**
Resolves #260

### CODE REVIEW

File: sdmetrics/single_table/new_row_synthesis.py [hunk 1]
```diff
@@ -70,30 +71,36 @@ def compute_breakdown(cls, real_data, synthetic_data, metadata=None,
...
+            if get_type_from_column_meta(field_meta) == 'datetime':
                 real_data[field] = pd.to_numeric(real_data[field])
                 synthetic_data[field] = pd.to_numeric(synthetic_data[field])
-                numerical_fields.append(field)
-            elif field_type == 'numerical':
-                numerical_fields.append(field)
+
+        try:
+            numerical_fields = cls._select_fields(metadata, ('numerical', 'datetime'))
+        except IncomputableMetricError:
```
Comment (amontanez24) [inline sdmetrics/single_table/new_row_synthesis.py:81]: Kind of unrelated by what does this method throw an `IncomputableMetricError`? It doesn't seem like it's computing a metric but maybe I'm misunderstanding

File: sdmetrics/single_table/new_row_synthesis.py [hunk 2]
```diff
@@ -70,30 +71,36 @@ def compute_breakdown(cls, real_data, synthetic_data, metadata=None,
...
+            numerical_fields = []
+
+        try:
+            categorical_fields = cls._select_fields(metadata, ('categorical', 'boolean'))
+        except IncomputableMetricError:
+            categorical_fields = []
```
Comment (amontanez24) [inline sdmetrics/single_table/new_row_synthesis.py:87]: very minor: but you could wrap these try/excepts in a method that gets the fields of a type, catches the error and returns an empty list if it does.

**Review by amontanez24 — APPROVED** [review]
Seems good! Some very minor comments

File: sdmetrics/single_table/new_row_synthesis.py (hunk 1 above)
Comment (katxiao) [reply sdmetrics/single_table/new_row_synthesis.py:81]: I think the original usage was like - if there are no fields of the valid type, we should throw an `IncomputableMetricError`. I do agree that doesn't make much sense to throw it in `_select_fields` - the method that is calling `_select_fields` should throw the error.

**Pull request merged**